src/api/rewrite.ts

```diff
@@ -1,21 +1,43 @@
 import { notifyError } from '../notifications/notifications';
 import { callSidecar } from './sidecar';
 
-export async function askForRewrite(selection: string): Promise<string> {
+export const REWRITE_MANNER: RewriteOptions['manner'][] = ['concise', 'elaborate', 'scholarly'];
+export interface RewriteOptions {
+  nChoices?: number;
+  manner?: 'concise' | 'elaborate' | 'scholarly';
+  temperature?: number; // between 0.7 and 0.9
+}
+
+export const DEFAULT_OPTIONS: RewriteOptions = {
+  nChoices: 3,
+  manner: 'concise',
+  temperature: 0.7,
+};
+
+export async function askForRewrite(selection: string, options?: RewriteOptions): Promise<string[]> {
   try {
     if (selection.trim() === '') {
-      return '';
+      return [''];
     }
 
-    const response = await callSidecar('rewrite', { text: selection });
+    options = {
+      ...DEFAULT_OPTIONS,
+      ...options,
+    };
+    const response = await callSidecar('rewrite', {
+      text: selection,
+      n_choices: options.nChoices,
+      temperature: options.temperature,
+      manner: options.manner,
+    });
 
     if (response.status === 'error') {
       notifyError('Rewrite error', response.message);
-      return '';
+      return [''];
```

Comment:
Shouldn't the function return `[]` when an error occurred?

Reply:
Good catch. We should actually return something more robust to the caller with a friendly message and an error status/flag.